services/data/src/links/RestAPILink/contentTypeForResource.test.ts

```diff
@@ -0,0 +1,19 @@
+import { contentTypeForResource } from './contentTypeForResource'
+
+describe('contentTypeForResource', () => {
+    it('returns text/plain for a string match', () => {
+        expect(contentTypeForResource('messageConversations/feedback')).toMatch(
+            'text/plain'
+        )
+    })
+    it('returns text/plain for a regex match', () => {
+        expect(
+            contentTypeForResource('messageConversations/g8mcplgksV2/priority')
```

Comment:
The resource here should actually be `messageConversations`, since resources should always be static.  A dynamic `id` of `{{id}}/priority` would need to be provided there.  It's unfortunate that these endpoints have nested resources, might be worth a rethink of how we do query resolution.

Reply:
> should actually be `messageConversations`

Very good point. I've think it's also important to note that while it _should_ indeed have been `messageConversations`, it _could_ still be `messageConversations/<ID>/priority`. We don't control that, and when using the engine imperatively I believe this works OK.

Since we don't control exactly how the resource is being constructed it'd probably be better to check the final `path`, i.e. the string which is produced by `queryToResourcePath` and match that with the list of resources that require specific content-types. That would require some further rearrangements, but could be done...

Reply:
In 2a5364a I have changed things around quite a bit and I am now matching against the path which is produced by `queryToResourcePath`. As such, I think we can resolve this conversation. Please do so if you agree @amcgee